### DISCUSSION TITLE
Get Signal.source.comment block without loading samples

### DISCUSSION THREAD

**Discussion by FloSchick** [danielhrisca/asammdf#514] [Q&A]
I‘m currently facing the difficulty that i have very large Signals of which i have to check the CAN number wich is located in the signal.source.comment attribute. Since my Signals are very large i just wanted to ask if anyone knows if i can get this comment without loading all the signal data which is quite slow?

**Comment by JulienGrv** [comment]
Hi FloSchick,

Yes you can. Here is a code snippet that should help you understand:

```python
from pathlib import Path

from asammdf import MDF


path = Path("my_file.mf4")

mdf = MDF(path)

# to get a list of ChannelGroup block source
cg_sources = [
    group.channel_group.acq_source
    for group in mdf.groups
]
# to get a list of Channel block source
cn_sources = [
    [
        channel.source
        for channel in group.channels
    ]
    for group in mdf.groups
]

```

**Comment by FloSchick** [comment]
@JulienGrv Thank you! That was exactly what i was looking for. Another question arises with this. I played around a bit with channel objects and I figured out, that when I for example modify Signal names, the internal functions such as filter() won’t find these new Signalnames until I save and reload the file. Is there a way to update the mdf object when channels have been modified?

**Reply by danielhrisca** [reply]
Hello @FloSchick , the `Signal` objects are decoupled from the `MDF` object (see https://asammdf.readthedocs.io/en/master/tips.html#data-coupling)

**Reply by FloSchick** [reply]
So there is no way around? My overall goal is to add the can number  into each signal name and then filter the mdf file with these new signal names. I found out that calling mdf.convert() updates the mdf attributes and then filtering with the edited names works. But it is also quiet slow.

**Reply by JulienGrv** [reply]
Probably you need to change the Channel block object name and not the Signal object name because of the decoupling as @danielhrisca mentioned

**Reply by danielhrisca** [reply]
> So there is no way around? My overall goal is to add the can number into each signal name and then filter the mdf file with these new signal names. I found out that calling mdf.convert() updates the mdf attributes and then filtering with the edited names works. But it is also quiet slow.

You can do this as a "dirty" trick: add a new entry in MDF.channels_db with the same group and channel index as the original signal
